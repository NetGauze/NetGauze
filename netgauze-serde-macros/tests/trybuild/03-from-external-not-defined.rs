#[allow(unused_imports)]
#[allow(unused_variables)]

use netgauze_serde_macros::LocatedError;
use netgauze_parse_utils::Span;

use nom::{
    error::ErrorKind,
    number::complete::be_u8,
    IResult,
};


#[derive(LocatedError, Eq, PartialEq, Debug, Clone)]
pub enum TestError {
    NomError(#[from_nom] ErrorKind),
    A(AError),
}

#[derive(Eq, PartialEq, Debug, Clone)]
pub enum AError {
    A
}

fn parse(buf: Span<'_>, ) -> IResult<Span<'_>, u8, LocatedTestError> {
    let (buf, ret) = nom::combinator::map_res(be_u8, |_| Err(AError::A))(buf)?;
    Ok((buf, ret))
}

fn main() {
    let span = Span::new(&[1]);
    let parsed = parse(span);
    assert_eq!(parsed, Err(nom::Err::Error(LocatedTestError::new(span, TestError::A(AError::A)))))
}