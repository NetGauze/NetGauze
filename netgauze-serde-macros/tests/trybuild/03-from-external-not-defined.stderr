error[E0277]: the trait bound `LocatedTestError<'_>: FromExternalError<netgauze_locate::BinarySpan<&[u8]>, AError>` is not satisfied
  --> tests/trybuild/03-from-external-not-defined.rs:26:22
   |
26 |     let (buf, ret) = nom::combinator::map_res(be_u8, |_| Err(AError::A))(buf)?;
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `FromExternalError<netgauze_locate::BinarySpan<&[u8]>, AError>` is not implemented for `LocatedTestError<'_>`
   |
   = help: the trait `FromExternalError<netgauze_locate::BinarySpan<&'a [u8]>, TestError>` is implemented for `LocatedTestError<'a>`
note: required by a bound in `map_res`
  --> $CARGO/nom-7.1.2/src/combinator/mod.rs
   |
   | pub fn map_res<I: Clone, O1, O2, E: FromExternalError<I, E2>, E2, F, G>(
   |                                     ^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `map_res`
